lib/src/resource/base.dart

```diff
@@ -36,3 +39,9 @@ abstract class Resource {
     throw UnimplementedError();
   }
 }
+
+abstract class ResourceRPCClient<T extends Client> {
+  abstract ClientChannelBase channel;
+
+  T get client;
+}
```

Comment:
this abstract class helps us enforce that each component has a channel and client.